fdbclient/NativeAPI.actor.cpp

```diff
@@ -2375,15 +2375,44 @@ Reference<WatchMetadata> DatabaseContext::getWatchMetadata(int64_t tenantId, Key
 }
 
 void DatabaseContext::setWatchMetadata(Reference<WatchMetadata> metadata) {
-	watchMap[std::make_pair(metadata->parameters->tenant.tenantId, metadata->parameters->key)] = metadata;
+	const WatchMapKey key(metadata->parameters->tenant.tenantId, metadata->parameters->key);
+	watchMap[key] = metadata;
+	// NOTE Here we do *NOT* update/reset the reference count for the key, see the source code in getWatchFuture
+}
+
+int32_t DatabaseContext::increaseWatchRefCount(const int64_t tenantID, KeyRef key) {
+	const WatchMapKey mapKey(tenantID, key);
+	if (watchCounterMap.count(mapKey) == 0) {
+		watchCounterMap[mapKey] = 0;
```

Comment:
Nit: assign 1 and return 1 here, to avoid access map again at line 2388.

Reply:
I think we don't even need this if block. If the item is not in the map, then `++watchCounterMap[mapKey]` will do the right thing.

Reply:
Yes. It is noted that for N3242, section 8.5.10/8.5.7 together with 23.5.4.3.2. Thought the behavior was no initialization.